server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/date/DateTimeUnitRounding.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.utils.date;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Interface for rounding time units in starTree
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface DateTimeUnitRounding {
```

Comment:
do we need an interface? Are we expecting more implementations?

Reply:
Actually hmmm we might have some custom implementation too in the future.

Reply:
This is basically to adapt the existing rounding + the new rounding date time unit classes. And yes, can extend for future as well.